DAP: 'invalid type: map, expected unit' on configurationDone response when adapter returns body={}

### Summary

`vscode-js-debug` (the DAP adapter Helix's `js-debug-dap` block is wired for) returns `body: {}` on the `configurationDone` response. Helix declares `ConfigurationDone::Result = ()` (unit type), so `serde_json::from_value::<()>(Value::Object({}))` fails with:

```
Debugger configuration failed: failed to parse: invalid type: map, expected unit
```

The error is shown to the user via `Editor::set_error` in the `Initialized` event handler, but the debug session continues (`set_active_client` runs regardless). The adapter has already accepted the request and returned `success: true`, so the debugger is actually running — the error is misleading, not fatal.

### Reproduction Steps

**`~/.config/helix/languages.toml`**

```toml
[[language]]
name = "javascript"

[language.debugger]
name = "js-debug-dap"
transport = "tcp"
port-arg = "{} 127.0.0.1"
command = "node"
args = ["/path/to/js-debug/src/dapDebugServer.js"]
quirks = { absolute-paths = true }

[[language.debugger.templates]]
name = "source"
request = "launch"
completion = [{ name = "main", completion = "filename", default = "index.js" }]
args = { type = "pwa-node", program = "{0}", skipFiles = ["<node_internals>/**"] }
```

**Adapter install** (Microsoft's prebuilt `js-debug-dap` release tarball, not the VS Code extension):

```bash
mkdir -p ~/lib/js-debug && cd ~/lib/js-debug
curl -L -o js-debug-dap.tar.gz \
  https://github.com/microsoft/vscode-js-debug/releases/download/v1.117.0/js-debug-dap-v1.117.0.tar.gz
tar -xzf js-debug-dap.tar.gz && rm js-debug-dap.tar.gz
# Entry point: ~/lib/js-debug/js-debug/src/dapDebugServer.js
```

**Steps:**

1. `echo 'console.log("hello")' > /tmp/index.js`
2. `hx /tmp/index.js`
3. Set a breakpoint (`<space>gb` on line 1)
4. `:debug-start source<ret>index.js<ret>`

**Expected:** debug session starts, status bar shows "Debugged application started".

**Actual:** status bar shows `Debugger configuration failed: failed to parse: invalid type: map, expected unit`. The debuggee actually starts, but the error message implies failure.

## Captured raw DAP response

Verified against the adapter directly (no Helix in the loop):

```json
{"seq": 2, "type": "response", "request_seq": 3, "command": "configurationDone", "success": true, "body": {}}
```

vscode-js-debug returns `body: {}` on every `configurationDone` response. This is technically non-DAP-compliant — the spec for `configurationDone`'s response has no body — but the same `{}` body shape is used by many other adapters (JDTLS, etc.) and Helix handles it elsewhere via `serde_json::Value` result types.


### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
helix_dap::transport [INFO] -> DAP {"type":"request","seq":3,"command":"configurationDone","arguments":{}}
helix_dap::transport [INFO] <- DAP {"seq":4,"type":"response","request_seq":3,"command":"configurationDone","success":true,"body":{}}
helix_dap::transport [INFO] <- DAP success in response to 3
helix_view::editor [ERROR] editor error: Debugger configuration failed: failed to parse: invalid type: map, expected unit
```

</details>


### Platform

Linux

### Terminal Emulator

alacritty 0.17.0 (94e7c887)

### Installation Method

source

### Helix Version

helix 25.07.1 (2868367c)

--- a/helix-dap/src/client.rs
+++ b/helix-dap/src/client.rs
@@ -463,8 +463,10 @@ impl Client {
             return Ok(());
         }
 
-        self.request::<requests::ConfigurationDone>(Some(requests::ConfigurationDoneArguments {}))
-            .await
+        self.call::<requests::ConfigurationDone>(Some(requests::ConfigurationDoneArguments {}))
+            .await?;
+
+        Ok(())
     }
 
     pub fn continue_thread(&self, thread_id: ThreadId) -> impl Future<Output = Result<Value>> {

